src/schedule/dma_multi_chan_domain.c

```diff
@@ -201,7 +202,7 @@ static void dma_multi_chan_domain_unregister(struct ll_schedule_domain *domain,
 
 	/* check if task should be unregistered */
 	if (!pipe_task->registrable)
-		return;
+		return -EINVAL;
```

Comment:
I dont know how failure will impact the scheduler, but this is probably important so should it be logged ?